src/BarracksServer/Database/Account.cs

```diff
@@ -330,5 +335,47 @@ public void Save()
 					BarracksServer.Instance.Database.SaveCharacter(character);
 			}
 		}
+
+		/// <summary>
+		/// Returns companion by companion id, or null if it doesn't exist.
+		/// </summary>
+		/// <param name="id"></param>
+		/// <returns></returns>
+		public Companion GetCompanionById(long id)
+		{
+			lock (_companions)
+				return _companions.FirstOrDefault(a => a.ObjectId == id);
+		}
+
+		/// <summary>
+		/// Returns list of all companions on account.
+		/// </summary>
+		/// <returns></returns>
+		public Companion[] GetCompanions()
+		{
+			lock (_companions)
+				return _companions.ToArray();
+		}
+
+		/// <summary>
+		/// Adds companion to account object and assigns index.
+		/// </summary>
+		/// <param name="companion"></param>
+		private void AddCompanion(Companion companion)
+		{
+			lock (_companions)
+			{
+				for (byte i = 1; i <= byte.MaxValue; ++i)
```

Comment:
While almost certainly not an issue in this case, C# conventions and our guidelines advocate for not using non-`int`s if not necessary for interoperability. `var` is also to be preferred.